server/controllers/cohort.js

```diff
@@ -188,33 +163,29 @@ const getCohortDetails = (req, resp) => {
         name
       });
     })
-    .catch(() =>
-      resp.status(400).send({
-        message:
-          'An error occurred while fetching the cohort data. Please try again.'
-      })
-    );
+    .catch(err => {
+      if (err instanceof NotFoundException) {
+        return resp.status(404).send();
```

Comment:
why not `res.sendStatus(404)`?

this could be event shorter:
`=> resp.sendStatus(err instanceof NotFoundException ? 404 : 400)`